earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -84,6 +118,7 @@ CreateEmptyDocument(const std::string &rootTagname) throw()
                                             0);// document type object (DTD)
     return doc;
```

Comment:
If I'm reading this correctly, the mutex is never unlocked unless an exception is thrown.

Reply:
Nevermind, I see the unlock in the destructor.